src/CameraScreen/CameraKitCameraScreenBase.js

```diff
@@ -103,13 +103,45 @@ export default class CameraScreenBase extends Component {
 
   getScannerOptions() {
     const scannerOptions = this.props.scannerOptions || {};
-    scannerOptions.offsetFrame = this.props.offsetForScannerFrame || OFFSET_FRAME;
-    scannerOptions.frameHeight = this.props.heightForScannerFrame || FRAME_HEIGHT;
-    if (this.props.colorForScannerFrame) {
-      scannerOptions.colorForFrame = processColor(this.props.colorForScannerFrame);
+    if (this.props.showFrame == true) {
+      if (this.props.frameHeight) {
+        scannerOptions.frameHeight = this.props.frameHeight
+      } else {
+        console.warn("frameHeight is required property, if you want to use scannerFrame set frameHeight value")
+      }
+      if (this.props.frameWidth) {
+        scannerOptions.frameWidth = this.props.frameWidth
+      } else {
+        console.warn("frameWidth is required property, if you want to use scannerFrame set frameWidth value")
+      }
+    }
+    scannerOptions.frameLeft = this.props.frameLeft || FRAME_LEFT
+    scannerOptions.frameTop = this.props.frameTop || FRAME_TOP
+
+    if (this.props.overlayColor) {
+      scannerOptions.overlayColor = processColor(this.props.overlayColor)
+    } else {
+      scannerOptions.overlayColor = processColor('rgba(0, 0, 0, 0.5)')
+    }
+
+    if (this.props.laserColor) {
+      scannerOptions.laserColor = processColor(this.props.laserColor)
+    } else {
+      scannerOptions.laserColor = processColor('white')
+    }
+    
+    if (this.props.frameColor) {
+      scannerOptions.colorForFrame = processColor(this.props.frameColor);
     } else {
       scannerOptions.colorForFrame = processColor("white");
+    }  
+    
+    if (this.props.surfaceColor) {
+      scannerOptions.surfaceColor = processColor(this.props.surfaceColor)
+    } else {
+      scannerOptions.surfaceColor = processColor("black")
     }
+    
```

Comment:
you actually pass an object to the native, but you are doing all the hard work to aggregate to an object instead of providing an API which is asking for an object

Reply:
only with optional values, as you see if ,when property are required we write it in logs
console.warn("frameWidth is required property, if you want to use scannerFrame set frameWidth value")